src/storage/store/engine.rs

```diff
@@ -269,7 +280,8 @@ impl ShardEngine {
 
                 match event {
                     Ok(hub_event) => {
-                        self.update_trie(hub_event, txn_batch)?;
+                        self.update_trie(hub_event.clone(), txn_batch)?;
+                        events.push(hub_event.clone());
```

Comment:
We're cloning three times in one function, can we avoid the clone by using references everywhere?